src/server/pbsd_init.c

```diff
@@ -256,14 +256,18 @@ init_server_attrs()
 {
 	resource_def    *prdef = NULL;
 	resource        *presc = NULL;
-	struct attribute attrib = {0};
-	int i = 0;
+	attribute 	temp_attr;
+	struct 		attribute attrib = {0};
+	int 		i = 0;
+
 
 	for (i=0; i<SRV_ATR_LAST; i++)
 		clear_attr(&server.sv_attr[i], &svr_attr_def[i]);
 
-	server.sv_attr[(int)SRV_ATR_scheduler_iteration].at_val.at_long =
-		PBS_SCHEDULE_CYCLE;
+	temp_attr.at_flags = ATR_VFLAG_SET;
+	temp_attr.at_val.at_long = PBS_SCHEDULE_CYCLE;
```

Comment:
@suresh-thelkar Please use at_decode function to set the values here and in all other places because decode is a common way of populating an attribute and if that changes in future we will not have to worry about these special cases. It will also set the flag for you.
Also, if you end up using decode, don't forget to call at_free as well because that will then free up memory inside the attribute (if allocated)

Reply:
Thanks for letting me know this. My initial idea was we should use decode functions only when we are getting data from external source i.e. either from database or from the user as input. One more thing what made me to use only set functions is most of the places in server code we are directly setting  value for an attribute of an object. But it seems it is better to use decode() followed by set functions as suggested by you. I have actually come up with a function named set_attr_svr() which takes care of all these and I am calling this wherever required.

And to correct all the places where we are not following this convention as of now which are outside the scope of this bug planning to raise an infrastructure RFE so that all the code we follow one approach. When we implement this we can actually enhance the newly written function set_attr_svr().